src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,180 +1,165 @@
 
   const venue = useSelector(currentVenueSelectorData);
-  const venueEvents = useSelector(venueEventsSelector) ?? [];
+  const venueEvents = useSelector(venueEventsSelector) ?? emptyEventsArray;
   const { worldUsers } = useWorldUsers();
 
   const { openUserProfileModal } = useProfileModalControls();
 
-  useEffect(() => {
-    const normalizedSearchQuery = searchQuery.toLowerCase();
-    if (!normalizedSearchQuery) {
-      setSearchResult({
-        rooms: [],
-        users: [],
-        events: [],
-      });
-      return;
+  const foundRooms = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
     }
 
-    const venueUsersResults = worldUsers.filter((user) =>
-      user.partyName?.toLowerCase()?.includes(normalizedSearchQuery)
+    /* @debt we really shouldn't be using the index as part of the key here, it's unstable.. but rooms don't have a unique identifier */
+    return (
+      venue?.rooms
+        ?.filter(
+          (room) =>
+            room.title.toLowerCase().includes(searchQuery) &&
+            room.type !== RoomTypes.unclickable
+        )
+        .map((room, index) => (
+          <NavSearchResult
+            key={`room-${room.title}-${index}`}
+            title={room.title}
+            description="Room"
+            image={room.image_url}
+            onClick={() => {
+              setSelectedRoom(room);
+              clearSearchQuery();
+            }}
+          />
+        )) ?? []
     );
+  }, [searchQuery, venue, clearSearchQuery]);
 
-    const venueEventsResults = venueEvents.filter((event) =>
-      event.name.toLowerCase().includes(normalizedSearchQuery)
-    );
+  const foundUsers = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    const roomsResults: Room[] =
-      venue?.rooms?.filter(
-        (room) =>
-          room.title.toLowerCase().includes(normalizedSearchQuery) &&
-          room.type !== RoomTypes.unclickable
-      ) ?? [];
+    return worldUsers
+      .filter((user) => user.partyName?.toLowerCase().includes(searchQuery))
+      .map((user) => (
+        <NavSearchResult
+          key={`user-${user.id}`}
+          title={user.partyName as string}
+          image={user.pictureUrl}
+          isAvatar={true}
+          onClick={() => {
+            openUserProfileModal(user);
+            clearSearchQuery();
+          }}
+        />
+      ));
+  }, [searchQuery, worldUsers, clearSearchQuery, openUserProfileModal]);
+
+  const foundEvents = useMemo<JSX.Element[]>(() => {
+    if (!searchQuery) {
+      return [];
+    }
 
-    setSearchResult({
-      rooms: roomsResults,
-      users: venueUsersResults,
-      events: venueEventsResults,
-    });
-  }, [searchQuery, venue, venueEvents, worldUsers]);
+    return venueEvents
+      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .map((event) => {
+        // @debt event can be related to the current venue as well but we don't receive its icon
```

Comment:
Can you explain the issue with this a bit more? Why don't we receive the icon/where are we trying to get it from/etc?

Reply:
today I finally found the place for currentVenue.icon. :see_no_evil: 
please, check the changes now: this is the way I'm getting events icons. https://github.com/sparkletown/sparkle/pull/1328/commits/45e88f45ce8010b8d16016235862348a4a6cbeca
if we couldn't find the room that is attached to the event, then we display the icon of the currentVenue.

<hr />

one more note. I think it might be necessary to open a new issue if there is no such.
The thing is that if we go into the child-venue, then we won't be able to search for events/rooms that are related to the parent venue. Because now it builds lists based on events/rooms that are related to the current venue only.

Reply:
> if we couldn't find the room that is attached to the event, then we display the icon of the currentVenue.

@denisdimitrov @mike-lvov Do you know if this is the correct image to be using here? I vaguely remember something about legacy images types on venues/etc (from the old Playa/etc days) that we are wanting to remove from the codebase/aren't relevant anymore. Not sure if the field used here is one of those, or if it's the one we want.

---

> one more note. I think it might be necessary to open a new issue if there is no such.
> The thing is that if we go into the child-venue, then we won't be able to search for events/rooms that are related to the parent venue. Because now it builds lists based on events/rooms that are related to the current venue only.

@sunny-viktoryia Interesting.. yeah. It might be worth opening a new issue about that, as the fix might be a bit larger/more convoluted/OOS than the scope of this PR.

We have `src/hooks/useConnectRelatedVenues.ts` which we use for building the schedule/etc. It might be that we want to end up using something similar here too.

@mike-lvov FYI: Any thoughts on this? Given where it is in the `NavBar`, this search feels like it should be a 'global search', yet based on the code, it's actually only searching for rooms/events within the current venue (`currentVenueSelectorData`, `venueEventsSelector`); so it's not going to work properly when used in most locations (aside from the main map)

Reply:
@0xdevalias @mike-lvov @sunny-viktoryia The icon of the current venue should be at `venue.host.icon` this is the so called "logo". 

>today I finally found the place for currentVenue.icon. 🙈

I don't think I've seen `venue.icon` called anywhere. If there is an icon/image in this field, we should probably remove the field, it's probably something legacy.

Reply:
sorry, I didn't mean not exact property names, but logic. I do get the icon from `venue.host.icon`. I just tried to explained the way I assign icons to events.

https://github.com/sparkletown/sparkle/blob/2574d407862199372cf770036bc25c76283e698d/src/components/molecules/NavSearchBar/NavSearchBar.tsx#L113